src/dft/backends/rocfft/commit.cpp

```diff
@@ -0,0 +1,427 @@
+            auto res =
+                rocfft_plan_description_set_data_layout(plan_desc, bwd_array_ty, fwd_array_ty,
+                                                        in_offsets.data(), // in offsets
+                                                        out_offsets.data(), // out offsets
+                                                        dimensions,
+                                                        in_strides.data(), //in strides
+                                                        bwd_distance, // in distance
+                                                        dimensions,
+                                                        out_strides.data(), // out strides
+                                                        fwd_distance // out distance
+                );
+            if (res != rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to set backward data layout.");
+            }
+
+            if (rocfft_plan_description_set_scale_factor(plan_desc, config_values.bwd_scale) !=
+                rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to set backward scale factor.");
+            }
+
+            rocfft_plan bwd_plan;
+            res = rocfft_plan_create(&bwd_plan, placement, bwd_type, precision, dimensions,
+                                     lengths.data(), number_of_transforms, plan_desc);
+            if (res != rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to create backward rocFFT plan.");
+            }
+            handles[1].plan = bwd_plan;
+
+            rocfft_execution_info bwd_info;
+            if (rocfft_execution_info_create(&bwd_info) != rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to create backward execution info.");
+            }
+            handles[1].info = bwd_info;
+
+            std::size_t work_buf_size;
+            if (rocfft_plan_get_work_buffer_size(bwd_plan, &work_buf_size) !=
+                rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to get backward work buffer size.");
+            }
+
+            if (work_buf_size != 0) {
+                void* work_buf;
+                if (hipMalloc(&work_buf, work_buf_size) != hipSuccess) {
+                    throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                         "Failed to get allocate backward work buffer.");
+                }
+                handles[1].buffer = work_buf;
+
+                if (rocfft_execution_info_set_work_buffer(bwd_info, work_buf, work_buf_size) !=
```

Comment:
From what I've read rocFFT can automatically create the work buffer if it is needed and free it. I think we should let it handle this automatically. Are we doing anything smarter than the automatic behavior?

Reply:
Do you have a reference for this?

Reply:
I was looking at https://rocfft.readthedocs.io/en/latest/allapi.html#_CPPv437rocfft_execution_info_set_work_buffer21rocfft_execution_infoPvK6size_t

Reply:
As discussed offline, this matched the cufft behaviour of of allocating the work area up front - https://docs.nvidia.com/cuda/cufft/index.html#function-cufftsetautoallocation